website/src/server/index.tsx

```diff
@@ -136,24 +135,38 @@ const httpServer = app.listen(port, host, backlog, () => {
 
 httpServer.timeout = timeout;
 
-// In development, it's common to stop or restart the server so we immediately end and close all
-// sockets when stopping the server instead of waiting for the requests to finish. In production,
-// we allow the requests a grace period to complete before ending and closing the sockets.
-const gracePeriod = process.env.NODE_ENV === 'development' ? 0 : httpServer.timeout;
-const server = stoppable(httpServer, gracePeriod);
+httpServer.keepAliveTimeout =
+  (10 * 60 + // Google Loadbalancing has an unconfigurable 10 minute timeout
+    20) * // Make our timeout 20 seconds longer than that ([This is recommended](https://cloud.google.com/load-balancing/docs/https#timeouts_and_retries))
+  1000; // Convert to milliseconds
 
 // Listen to HTTP server error events and handle shutting down the server gracefully
 let exitSignal: ShutdownSignal | null = null;
 let httpServerError: Error | null = null;
+let ready = true;
 
-server.on('error', (error) => {
+const readiness = new Koa();
+
+readiness.use(async function (ctx) {
+  if (ready) {
+    ctx.response.status = 200;
+    ctx.body = 'ready';
+  } else {
+    ctx.response.status = 503;
+    ctx.body = 'shutting down';
+  }
+});
+
+app.use(mount('/ready', readiness));
```

Comment:
koa-mount also supports mounting middleware directly so there's no need to create a whole Koa object.
```suggestion
app.use(mount('/ready', function readiness(ctx) {
  if (ready) {
    ctx.response.status = 200;
    ctx.body = 'ready';
  } else {
    ctx.response.status = 503;
    ctx.body = 'shutting down';
  }
}));
```